src/server/transaction.cc

```diff
@@ -1160,26 +1162,16 @@ OpStatus Transaction::WatchInShard(ArgSlice keys, EngineShard* shard) {
   return OpStatus::OK;
 }
 
-void Transaction::UnwatchShardCb(ArgSlice wkeys, bool should_expire, EngineShard* shard) {
-  if (should_expire) {
-    auto lock_args = GetLockArgs(shard->shard_id());
-    shard->db_slice().Release(Mode(), lock_args);
-
-    unsigned sd_idx = SidToId(shard->shard_id());
-    auto& sd = shard_data_[sd_idx];
-    sd.local_mask |= EXPIRED_Q;
-    sd.local_mask &= ~KEYLOCK_ACQUIRED;
-    shard->blocking_controller()->FinalizeWatched(wkeys, this);
-    DCHECK(!shard->blocking_controller()->awakened_transactions().contains(this));
-  }
-
-  // Need to see why I decided to call this.
-  // My guess - probably to trigger the run of stalled transactions in case
-  // this shard concurrently awoke this transaction and stalled the processing
-  // of TxQueue.
-  shard->PollExecution("unwatchcb", nullptr);
-
```

Comment:
All that happened previously for non-expired tx was this poll execution

It seems like if the tx is not expired, it will invariably follow up with a hop, so the balance of Polls/Executions will remain intact

Reply:
There's been a few blocking related issues lately, could you maybe reference the specific bug that this fixes? Or a unit test for the issue